backends/p4tools/modules/smith/scripts/compilation-test.sh

```diff
@@ -47,7 +63,34 @@ TARGET=$6
 
 for i in $(seq 1 $NUM_ITERATIONS); do
     echo "Generating program $i"
-    $SMITH_BIN --target $ARCH --arch $TARGET --seed $i $TEST_DIR/out.p4
+    # TODO(zzmic): Revert out_$i.p4 to out.p4
+    # Generate different programs with different seeds (to inspect the behavior of the generation and compilation process in a finer/smaller granularity).
+    echo "$SMITH_BIN --arch $ARCH --target $TARGET --seed $i $TEST_DIR/out_$i.p4"
+    $SMITH_BIN --arch $ARCH --target $TARGET --seed $i $TEST_DIR/out_$i.p4
     # TODO: Do not compile until we have stabilized.
-    # $COMPILER_BIN $TEST_DIR/out.p4
+    # TODO(zzmic): Revert out_$i.p4 to out.p4
+    # If the arch is "pna", the arch itself shall be specified.
+    # Otherwise, the compilation would trigger
+    # "Arch '<null>' not supported by P4Runtime serializer"
+    if [ "$ARCH" == "pna" ]; then
```

Comment:
Rather than hardcoding a check here, make this part of the CMake portion of the tests. You can specify the compiler binary with extensions `--arch` and `--target` there.